### PR TITLE
CB-9011 Adding Buildbot feature support in medic

### PR DESCRIPTION
**Pull request by dblotsky**
Following are notable changes.

- Added commit-triggered builds.
- Now using Buildbot Codebases for each codebase; no more downloading repo configs.
- Made code more modular and less repetitive.
- Added GitHub hook support for test installations of Medic (the Apache Infrastructure installation of Medic has a different `master.conf`).
- Periodic builds no longer run if no changes are present.

### CODE REVIEW

File: buildbot-conf/cordova-repos.json [hunk 1]
```diff
@@ -1,457 +1,285 @@
...
-        "codebases": {
-            "asf":  {
-                "repo":   "https://git-wip-us.apache.org/repos/asf/cordova-medic.git",
-                "branch": "master"
-            },
-            "github":  {
-                "repo":   "https://github.com/apache/cordova-medic.git",
-                "branch": "master"
-            }
+        "default_repository": "msot",
```
Comment (alsorokin) [inline buildbot-conf/cordova-repos.json]: The default repository is "msot" but "repositories" contain only "asf" and "github".
Is that intended?

Comment (dblotsky) [inline buildbot-conf/cordova-repos.json]: Nice catch! That is an outright oversight.

**Comment by nikhilkh** [comment]
This is a significant improvement in our BuildBot capability! Let's get this in and up on Apache CI.

**Comment by purplecabbage** [comment]
+1

**Pull request merged**